src/SimplygonFunctionApp/AzRemeshFn.cs

```diff
@@ -1,195 +1,390 @@
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            var data = await ProcessParameters(req);
+
+            log.LogInformation($"Input parameters = {data}");
+
+            var zipDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()); // Zip extraction creates the directory
+            var outputDir = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()));
+
+            log.LogInformation($"zip dir = {zipDir}");
+
+            HttpResponseMessage response = null;
+
+            using (var http = new HttpClient())
+            {
+                log.LogInformation($"HTTP GET = {data.InputZipUri}");
+
+                try
+                {
+                    response = await http.GetAsync(data.InputZipUri);
+                }
+                catch (Exception ex)
+                {
+                    log.LogError(ex, $"Request to {data.InputZipUri} Failed");
+                }
+            }
+
+            using (var za = new ZipArchive(await response.Content.ReadAsStreamAsync(), ZipArchiveMode.Read))
+            {
+                za.ExtractToDirectory(zipDir, true);
+            }
+
+            string remeshedFile;
+
+            using (var sg = Simplygon.Loader.InitSimplygon(out var errorCode, out var errorMessage))
+            {
+                if (errorCode != Simplygon.EErrorCodes.NoError)
+                    return new BadRequestObjectResult($"Failed! {errorCode} - {errorMessage}");
+
+                // Find the first .glTF file in the input and process it.
+                var fileToProcess = Directory.GetFiles(zipDir, "*.glTF", SearchOption.AllDirectories)
+                    .FirstOrDefault();
+                if (string.IsNullOrEmpty(fileToProcess))
+                {
+                    const string BadInputError = ".glTF input not found in zip archive.";
+                    log.LogError(BadInputError);
+                    return new BadRequestObjectResult($"Failed! {BadInputError}");
+                }
+
+                remeshedFile = await RunRemeshingWithMaterialCastingAsync(sg, log, fileToProcess, ToOutputPath(outputDir, fileToProcess), data.OnScreenSize);
+            }
+
+            log.LogInformation($"Processed {remeshedFile}");
+
+            var stream = new MemoryStream();
+
+            using (var outputZip = new ZipArchive(stream, ZipArchiveMode.Create, true))
+            {
+                outputZip.CreateEntryFromDirectory(outputDir.FullName);
+            }
+
+            stream.Position = 0;
+            log.LogInformation("Cleaning up ... ");
+
+            // Cleanup by deleting the directories
+            try {
+                outputDir.Delete(true);
+                // Directory.Delete(zipDir, true); // this likely throws an exception (dangling file pointer?) but the exception is not caught by the `catch` block

```

Comment:
Good practice as well: no dead code in code!
If you want to place dead code, you hace to justify why.